### DISCUSSION TITLE
Can I ignore changes to files that are already part of the tree?

### DISCUSSION THREAD

**Discussion by BatmanAoD** [jj-vcs/jj#4587] [Q&A]
I have some files that are part of my git repo, and which, due to some legacy testing setup that I haven't had time to untangle yet, are changed every time I run my unit tests; but those updates _should not_ actually be committed.

I've tried adding the files to `.gitignore`, but `jj` still records the changes, which makes sense since [git does too](https://stackoverflow.com/q/1274057/1858225). (I can't actually get git to ignore the files even using `--skip-worktree` _or_` --assume-unchanged`, which is...troubling.) I don't think the new auto-track option would solve this since it's a glob pattern so it can't easily be negated, and I also don't know whether it applies only to new files or to already-tracked files (the release notes just link to the issue, and I didn't read the whole conversation there).

So is there any way to tell `jj` "ignore changes to these files by default"?

**Comment by martinvonz** [comment]
No, we don't have such a feature. It could be implemented in a similar way as the auto-track option, if we think it's worth the added complexity.

**Reply by BatmanAoD** [reply]
I'll think about it and post here again if I run into a situation that seems more legitimate than what I described above.

**Reply by anka-213** [reply]
Another situation where I use `--skip-worktree` is when I want to have local changes to a config file that doesn't have support for a `.local.cfg` version or similar. Is there a standard solution to this?

**Reply by jennings** [reply]
@anka-213 See this FAQ: [How can I avoid committing my local-only changes to tracked files?](https://jj-vcs.github.io/jj/latest/FAQ/#how-can-i-avoid-committing-my-local-only-changes-to-tracked-files).

The trick is to keep your local-only change in a commit "to the side of" instead of "on top of" the branch you're working on.

**Comment by scott2000** [comment]
You could maybe use a sparse working copy to manually exclude the files from snapshotting, but if any new files are added in any parent directory of the files, then they would have to be manually added to the sparse checkout (if I understand correctly).

**Comment by psowinski** [comment]
Please consider implementing something like --assume-unchanged, there are real scenarios when we don't want to commit a file and keeping changes aside is not always convenient
